kyverno-pss-policy-exceptions/README.md

```diff
@@ -0,0 +1,167 @@
+Giant Swarm:
+
+- creation of `PolicyExceptions` for our applications which require them and can't be made compliant
+  - Shield has already made many workloads compliant so they don't require exceptions. Others can still be made compliant.
+  - this work could be done in parallel with removal / gating of PSP resources which will need to be removed in post-v1.25 clusters anyway
+- (Shield) creation of assistive guides. This is already in progress in parallel, but requires agreement on this RFC to be finalized
+
+Customers:
+
+- creation of `PolicyExceptions` for customer workloads
+  - this effort depends significantly on the customer's current model of PSP usage and application behavior
+    - for example, we have customers who centrally control all PSPs and customers who allow teams to control their own. Effort for the second is likely to be higher, also from a future management perspective
+
+Shield's experience with early Kyverno adopters is that organizations have drastically different ability and willingness to enforce security policies.
+We will provide resources based on what we already know from early adopters, what we learn from Giant Swarm's migration, and the unknown needs that we will inevitably learn from customers as they begin the move.
+
+### Migration Path
+
+More detailed information on the migration path will be published after agreement is reached on this RFC. To help understand the path to PSS adoption, the high-level migration process would be:
+
+1. Kyverno and PSS policies (audit) are added to an upcoming release. This could be a new v19 minor.
+    - Result: customers now have information about non-compliant workloads
+1. Customers begin creating `PolicyExceptions` for workloads which require them
+    - Shield provides docs and support
+    - Optionally: customers change individual policies to `enforce` as compliance is achieved
+    - Optionally: customers can elect to completely disable PSPs
+1. Kubernetes 1.25 release is published, containing policies in `enforce` mode by default
+    - Result: PSPs are removed
+    - Result: remaining non-compliant workloads will be rejected
+    - Result: customers must deploy exceptions for remaining workloads or grant more widely-scoped exceptions (e.g. disabling enforcement of certain policies)
+
+## Alternatives
+
+Multiple alternatives have been considered and attempted over the past two years, including:
+
+### Exceptions via Labels / Annotations
+
+This approach allows resources to opt themselves out of policies by applying certain labels or annotations to the resource.
+
+The advantage of this approach was that it made self-service very easy for developers, and was for some time the only technically feasible way to handle self-service exceptions in Kyverno logic.
+
+There are several major drawbacks, the largest being that labels and annotations are not RBAC-controlled, so this allows most users the ability to exempt themselves from any policy without administrator review.
```

Comment:
I am not sure if this is good in all clusters (like prod), can we disable by cluster?

Reply:
Label / Annotation-based exceptions are an alternative we do not want to pursue for several reasons. I presented it here for context and transparency